ecs-cli/modules/config/readwriter_v1.go

```diff
@@ -66,48 +67,140 @@ func readINI(dest *Destination) (*CLIConfig, map[interface{}]interface{}, error)
+	}
+
+	if to, configMap, errINI := readINI(rdwr.destination); errINI == nil {
 		// File is INI
 		return to, configMap, nil
-	} else if errINI != nil {
-		// Return the ini error
-		// This will return a parsing error for ini (format error)
-		return nil, nil, errINI
-	} else {
-		// If yaml through an error, but ini didn't throw an error- return the yaml error
-		return nil, nil, errYAML
 	}
+
+	if _, err := os.Stat(configPath); err != nil {
+		// Cluster config file exists, but it wasn't INI, so that means it was YAML but there was a parse error
+		return nil, nil, errYAMLConfig
+	}
+
+	// if no configs exist, we return empty objects
+	return cliConfig, configMap, nil
+
+}
+
+func processProfileMap(profileKey string, profileMap map[interface{}]interface{}, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
+	if profileKey == "" {
+		var ok bool
+		if profileKey, ok = profileMap["default"].(string); !ok {
+			return errors.New("Format issue with profile config file; expected key not found.")
+		}
+	}
+	profile, ok := profileMap["ecs_profiles"].(map[interface{}]interface{})[profileKey].(map[interface{}]interface{})
+	if !ok {
+		return errors.New("Format issue with profile config file; expected key not found.")
+	}
+
+	configMap[awsAccessKey] = profile[awsAccessKey]
+	configMap[awsSecretKey] = profile[awsSecretKey]
+	if cliConfig.AWSAccessKey, ok = profile[awsAccessKey].(string); !ok {
+		return errors.New("Format issue with profile config file; expected key not found.")
+	}
+	if cliConfig.AWSSecretKey, ok = profile[awsSecretKey].(string); !ok {
+		return errors.New("Format issue with profile config file; expected key not found.")
+	}
+
+	return nil
+
+}
+
+func processClusterMap(clusterConfigKey string, clusterMap map[interface{}]interface{}, configMap map[interface{}]interface{}, cliConfig *CLIConfig) error {
+	if clusterConfigKey == "" {
+		var ok bool
+		if clusterConfigKey, ok = clusterMap["default"].(string); !ok {
+			return errors.New("Format issue with cluster config file; expected key not found.")
+		}
+	}
+	cluster, ok := clusterMap["clusters"].(map[interface{}]interface{})[clusterConfigKey].(map[interface{}]interface{})
+	if !ok {
+		return errors.New("Format issue with cluster config file; expected key not found.")
+	}
+
+	configMap[clusterKey] = cluster[clusterKey]
+	configMap[regionKey] = cluster[regionKey]
+	if cliConfig.Cluster, ok = cluster[clusterKey].(string); !ok {
+		return errors.New("Format issue with cluster config file; expected key not found.")
+	}
+	if cliConfig.Region, ok = cluster[regionKey].(string); !ok {
+		return errors.New("Format issue with cluster config file; expected key not found.")
+	}
+
+	// Prefixes
+	// ComposeProjectNamePrefix
+	if _, ok := cluster[composeProjectNamePrefixKey]; ok {
+		configMap[composeProjectNamePrefixKey] = cluster[composeProjectNamePrefixKey]
+		if cliConfig.ComposeProjectNamePrefix, ok = cluster[composeProjectNamePrefixKey].(string); !ok {
```

Comment:
Once again, there is actually a reason for this annoying complex code. It is the issue that its possible for composeProjectPrefixKey to both exist and be empty, and for it to not exist as well.